draft-ietf-jsonpath-base.md

```diff
@@ -550,7 +550,9 @@ The syntax and semantics of segments are defined in {{segments-details}}.
 
 In this document, the semantics of a JSONPath query define the
 required results and do not prescribe the internal workings of an
-implementation.
+implementation.  This document may describe semantics in a procedural
+step-by-stap fashion, but such descriptions are non-normative and do not
```

Comment:
Well, the descriptions are meant to be normative, but the way they are described is not the only way of implementing them.

Reply:
Something like "… step-by-step function, which are normative in the sense that any implementation MUST produce an identical result, but not in the sense that implementors are required to use the same algorithms" ?